libraries/helpers.rb

```diff
@@ -23,4 +23,13 @@ module ChefHelpers
   def self.secret_files
     %w{pivotal.cert  pivotal.pem  webui_priv.pem  webui_pub.pem  worker-private.pem  worker-public.pem}
   end
+
+  def self.symbolize_keys_deep!(h)
+    Chef::Log.debug("#{h.inspect} is a hash with string keys, make them symbols")
+    h.keys.each do |k|
+      ks    = k.to_sym
+      h[ks] = h.delete k
+      symbolize_keys_deep! h[ks] if h[ks].kind_of? Hash
```

Comment:
I think I want to enforce this to be a Hash. If we're a Mash or attribute, we're going to have the "keys aren't actually symbols" problem.